### ISSUE TITLE
Brave adblock not working with Fanboy's Annoyance List

### ISSUE THREAD

**Issue by m040601** [qutebrowser/qutebrowser#6469]
There is already an issue "Brave adblock not working"
https://github.com/qutebrowser/qutebrowser/issues/6460
which got closed and redirected to, "adblock not working in youtube"
https://github.com/qutebrowser/qutebrowser/issues/6399

I too seem to be having this problem, but not exactly the same.
Since these issues, seem to be too much focused on youtube, and after extensive testing mine seems to be different,I decided to open a new one

Please tell me if I'm wrong.

> What kind of ads do you get exactly?

In my case, I noticed it not in youtube, but in adblock failure to load and parse an **extra adblock list** and **use it effectively.**

Fanboy's Annoyance List, https://easylist.to/

> Fanboy's **Annoyance** List blocks Social Media content, in-page pop-ups and other annoyances; thereby substantially decreasing web page loading times and uncluttering them. EasyList Cookie List and Fanboy's Social Blocking List are already **included**, there is no need to subscribe to them if you already have Fanboy's Annoyance List.
 
> Fanboy's **Social** Blocking List **solely** removes Social Media content on web pages such as the Facebook like button and other widgets.

Raw text version, https://secure.fanboy.co.nz/fanboy-annoyance.txt

I'm very frugal with customizations and avoid like the plague ticking extra options and redundant extra lists, but this one is in fact very good.
This is one of the very few  **extra things** I added to the defaults.

```
config_version: 2                                                                   
settings:                                                                           
  content.autoplay:                                                                 
    global: false                                                                   
  content.blocking.adblock.lists:                                                   
    global:                                                                         
    - https://easylist.to/easylist/easylist.txt                                     
    - https://easylist.to/easylist/easyprivacy.txt                                  
    - https://secure.fanboy.co.nz/fanboy-annoyance.txt                              
  content.blocking.method:                                                          
    global: auto                                                                 
  content.dns_prefetch:                                                             
    global: false         
```                                                           

I know this fanboy-annoyance list is good, because itworks very well in Firefox, and it's usually one of the few extra options I tick on ublock origin.

> How to reproduce

Load any major media site with "share with" buttons.
- Firefox with ublock origin, tick "fanboy annoyances"
- Qutebrowser with extra braveblock list "https://secure.fanboy.co.nz/fanboy-annoyance.txt"

Compare. On Firefox, this extra list effectivly cleans the web from all those "cookie consents" and "share with" buttons. On Qutebrowser they are still there.

> What does the adblock: ... line say in :version?

adblock: 0.4.4
> What does :set content.blocking.method say?

auto
> Did you run :adblock-update recently?

of course. After I run it, I make sure to **watch carefully the INFO messages** and check that it was effectively downloaded and parsed.

> Downloading adblock filter lists...
> hostblock: Read 80586 hosts from 1 sources.
> **braveadblock**: Filters **successfully** read from **3 source**
> 

I even do
 `wget https://secure.fanboy.co.nz/fanboy-annoyance.txt`
 on the terminal to make sure there is no network failure.

I've tested this over and over, comparing qutebrowser and firefox, both my standard installation and fresh not customized ones.

I've even created a local list with just a few simple rules to block some  domains, containing

`||https://www.somesite.com^
`
And I feed to qutebrowser braveblock  lists array (not hostblock lists)

`http://localhost/mybraveblocklist.txt
`
and make sure it was beeing parsed and enforced.
And indeed, I could see the rules being applied and it's domains beeing blocked

`00:45:12 ERROR    webview    webenginetab:_error_page_workaround:1485 Load error: ERR_BLOCKED_BY_CLIENT
`

> reproduce with --temp-basedir? So the ads are blocked there?

I made extensive use of
`qutebrowser -T --debug --logfilter network`
to follow in real time on the terminal whats happening.

I watch the log messages seeing the usualads (google analytics etc) being blocked by the **default lists**.
`"https://easylist.to/easylist/easylist.txt", "https://easylist.to/easylist/easyprivacy.txt"`

But qutebrowser always fails to block these "social media" icons and enforce this **extra fanboy list**.
So these extra ads, that should be blocked by fanboy's list,  **are not blocked  with --temp-basedir.**

```
qutebrowser v2.2.1
Backend: QtWebEngine 5.15.3, Chromium 87.0.4280.144
Qt: 5.15.2

CPython: 3.9.5
PyQt: 5.15.4

sip: 6.1.0.dev2104271705
colorama: 0.4.4
jinja2: 3.0.0
pygments: no
yaml: 5.4.1
adblock: 0.4.4
PyQt5.QtWebEngineWidgets: yes
PyQt5.QtWebEngine: 5.15.4
PyQt5.QtWebKitWidgets: no
pdf.js: no
sqlite: 3.35.5
QtNetwork SSL: OpenSSL 1.1.1k  25 Mar 2021

Style: QFusionStyle
Platform plugin: xcb
OpenGL: X.Org, 3.0 Mesa 21.1.0
Platform: Linux-5.12.4-arch1-2-x86_64-with-glibc2.33, 64bit
Linux distribution: Arch Linux (arch)
Frozen: False
Imported from /usr/lib/python3.9/site-packages/qutebrowser
Using Python from /usr/bin/python3
Qt library executable path: /usr/lib/qt/libexec, data path: /usr/share/qt

Paths:
cache: /tmp/qutebrowser-basedir-06_x0ntu/cache
config: /tmp/qutebrowser-basedir-06_x0ntu/config
data: /tmp/qutebrowser-basedir-06_x0ntu/data
runtime: /tmp/qutebrowser-basedir-06_x0ntu/runtime
system data: /usr/share/qutebrowser

Autoconfig loaded: yes
Config.py: no config.py was loaded
```

Extra comment about youtube.
I usually dont notice it much on youtube, because I almost never load videos on the youtube page with qutebrowser. I spawn mpv/youtube-dl to watch the videos.

So I cant say whether ads on youtube with qutebrowser are worse or better than before.
They are there right now with qutebrowser.

Funny thing, a Firefox with ublock origin seems to block youtube ads completely.
Now and then comments like,

> Ads showing up on youtube now

show up in Reddit and Qutebrowser issues. But I never see saw them. Firefox with a proper and frugal ublock origin configured seems solid to me.

If I recall it from last year, qutebrowser always blocked youtube worse. It always tried to play a few seconds of the ad.
I cant remember exactly, but I had to use an extra Greasemonkey stream to effectivly block youtube **video ads** with qutebrowser.
I think it was

> Auto Close YouTube 
> http://fuzetsu.acypa.com

**Comment by m040601** [comment]
the compiler wrote:

> .... not much I can say without a link to an affected page, but I'm guessing it's because cosmetic filtering isn't implemented so far, see #5754.

Ah so that seems to explain it.

I guess I was expecting "too much" from qutebrowser's braveblock.
I was expecting to be a 100% equivalent to a mature solution like ublock origin.
And therefore was expecting get the same exact blocking results in Qutebrowser as in Firefox.

So we arent there yet. Not that I'm complaining. I've been satisfied with qutebrowser even with the simple "hostblock" solution. Hope the "braveblock" can one day match ublock origin.


>  not much I can say without a link to an affected page

This affects any page. And the only way to properly test it is to compare firefox and qutebrowser with this same "Fanboy's Annoyance" list.

Example.
Remember to do this **with a fresh profile before you visit, and click and dismiss anything.** 
This is what you see the first time you visit the bbc.com or nzz.ch
https://imgur.com/a/G6dwWgH


Now start **again with a fresh profile before you visit, and click and dismiss anything.**
Now add the list **before** you visit any major media site.
Go again to bbc.com or nzz.ch
You'll notice that although it's like **your first visit ever** to those sites, the banners arent there. It's as if someone had clicked them away for you.

Same thing for annoying "share with social media" bars.
https://imgur.com/a/h8F9OZ3

**Comment by The-Compiler** [comment]
Duplicate of #5754
